File: Backend/manga-world/src/main/java/com/amakakeru/mangaworld/entity/ProfilePicture.java
package com.amakakeru.mangaworld.entity;

import jakarta.persistence.*;
import lombok.Data;

import java.io.Serializable;

@Data
@Entity
@Table(name = "profile_picture")
public class ProfilePicture implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "picture_id")
    private Long pictureId;

    @Lob
    @Column(name = "picture", nullable = false)
    private byte[] picture;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", unique = true)
    private User user;
}

File: Frontend/manga-world/src/Model/ProfilePicture.ts
import User from "./user";

class profilePicture {
    pictureId: number;
    picture: string;
    user: User;

    constructor(pictureId: number, picture: string, user: User) {
        this.pictureId = pictureId;
        this.picture = picture;
        this.user = user;
    }
}

export default profilePicture;  